flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -268,6 +278,10 @@ public LogicalSlot getAssignedResource() {
 		return assignedResource;
 	}
 
+	public Map<IntermediateResultPartitionID, ResultPartitionDeploymentDescriptor> getProducedPartitions() {
+		return producedPartitions;
+	}
```

Comment:
We could hide this implementation detail by offering a `Optional<ResultPartitionDeploymentDescriptor> getProducedPartition(IntermediateResultPartitionID intermediateResultPartitionId)`

Reply:
The return type could even be `Optional<ShuffleDeploymentDescriptor>` if I'm not mistaken.